File: programmers/Lv_2/KimHyunSeo/124나라의숫자.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(int n) {
	string answer = "";
	while (n > 0) {
		if (n % 3 == 0)
		{
			answer.insert(0, "4");
			n = n / 3 - 1;
		}
		else if (n % 3 - 1 == 0)
		{
			answer.insert(0, "1");
			n /= 3;
		}
		else
		{
			answer.insert(0, "2");
			n /= 3;
		}
	}
	return answer;
}

File: programmers/Lv_2/JeongJunKwon/124나라의 숫자.java
class Solution {
  public String solution(int n) {
      String answer = "";
      while(n > 0) {
          int mod = n % 3;
          switch(mod) {
              case 0:
                  answer = "4" + answer;
                  n -= 3;
                  break;
              case 1:
                  answer = "1" + answer;
                  n -= 1;
                  break;
              case 2:
                  answer = "2" + answer;
                  n -= 2;
                  break;
          }
          n /= 3;        
          }
      return answer;
  }
}